plugins/CoreHome/vue/src/Matomo/Matomo.ts

```diff
@@ -6,25 +6,111 @@
  */
 
 import Periods from '../Periods/Periods';
-
-let originalTitle: string;
+import { translate } from '../translate';
 
 const { piwik, broadcast, piwikHelper } = window;
 
+type ComparisonsStoreLike = {
+  getSegmentComparisons: () => Array<{params: { segment: string }, title: string, index: number}>;
+};
+
+type ReportingMenuStoreLike = {
+  findSubcategory: (categoryId: string, subcategoryId: string) => {
+    category?: { name?: string };
+    subcategory?: { name?: string };
+  };
+  fetchMenuItems: () => Promise<unknown>;
+};
+
 piwik.helper = piwikHelper;
 piwik.broadcast = broadcast;
 
-piwik.updateDateInTitle = function updateDateInTitle(date: string, period: string) {
-  if (!$('.top_controls #periodString').length) {
-    return;
+async function getReportingMenuStore(): Promise<ReportingMenuStoreLike|undefined> {
+  const coreHome = (window as unknown as {
+    CoreHome?: { ReportingMenuStore?: ReportingMenuStoreLike };
+  }).CoreHome;
+
+  return coreHome?.ReportingMenuStore;
+}
+
+function getComparisonsStore(): ComparisonsStoreLike|undefined {
+  const coreHome = (window as unknown as
+    { CoreHome?: { ComparisonsStoreInstance?: ComparisonsStoreLike } }).CoreHome;
+  return coreHome?.ComparisonsStoreInstance;
+}
+
+function getActiveSegmentLabel(segment?: string): string|undefined {
+  if (typeof segment !== 'string') {
+    return undefined;
+  }
+
+  const trimmedSegment = segment.trim();
+  const comparisonsStore = getComparisonsStore();
+
+  if (comparisonsStore) {
+    const comparisons = comparisonsStore.getSegmentComparisons();
+    if (!trimmedSegment && comparisons.length) {
+      return comparisons[0].title;
+    }
+
+    const found = comparisons.find(
+      (comparison) => comparison.params.segment === segment,
+    );
+    if (found) {
+      return found.title;
+    }
+  }
+
+  if (!trimmedSegment) {
+    return translate('SegmentEditor_DefaultAllVisits');
   }
 
-  // Cache server-rendered page title
-  originalTitle = originalTitle || document.title;
+  const segmentationTitle = document.querySelector('.segmentEditorPanel .segmentationTitle');
+  const fallbackName = segmentationTitle?.textContent?.trim();
+  if (fallbackName) {
+    return fallbackName;
+  }
+
+  return translate('SegmentEditor_CustomSegment');
+}
+
+piwik.updateTitle = async function updateTitle(
+  date: string,
+  period: string,
+  c: string,
+  s: string,
+  segment?: string,
+) {
+  let categoryName = '';
+  let subcategoryName = '';
+  let dateString = '';
+  if (period !== '' && date !== '') {
+    dateString = `${Periods.parse(period, date).getPrettyString()} `;
```

Comment:
The ending space doesn’t seem to be needed.

```suggestion
    dateString = Periods.parse(period, date).getPrettyString();
```

Reply:
Good spotting on this one 
Cheers